templates/publishArtifacts/publishArtifacts.yml

```diff
@@ -23,9 +23,20 @@ steps:
       ArtifactName: AcceptanceTests
     condition: ne(variables['AcceptanceTestsPath'], '')
 
+  - powershell: |
+     $infrastructurePath = (Resolve-Path $(Build.SourcesDirectory)\Infrastructure -ErrorAction Continue).path
+     if ($null -eq $infrastructurePath){
+     Write-Output "##vso[task.logissue type=warning]Infrastructure files not found"
+     }
+     else{
+     Write-Output "##vso[task.setvariable variable=Infrastructure;]$infrastructurePath"
+     Write-Output ("Infrastructure file path {0}" -f $infrastructurePath)
+     }
+    displayName: Resolve Infrastructure files path
```

Comment:
indentation is a bit off?
could also put in a comment to explain why we're doing this sometimes and sometimes not?